src/repositories/skill-category-repository.ts

```diff
@@ -6,3 +7,85 @@ export const getAllByFilters = async (where: Prisma.skill_categoriesWhereInput)
     where,
   })
 }
+
+export const getById = async (id: number) => {
+  return await prisma.skill_categories.findUnique({
+    where: {
+      id,
+    },
+  })
+}
+
+export const getByName = async (name: string) => {
+  return await prisma.skill_categories.findFirst({
+    where: {
+      name,
+    },
+  })
+}
+
+export const create = async (data: SkillCategory) => {
+  const currentDate = new Date()
+  const newSkillCategory = await prisma.skill_categories.findFirst({
+    orderBy: { sequence_no: "desc" },
+  })
+  const nextSequenceNo =
+    newSkillCategory?.sequence_no !== null && newSkillCategory?.sequence_no !== undefined
+      ? newSkillCategory.sequence_no + 1
+      : 1
+  const newCategory = await prisma.skill_categories.create({
+    data: {
+      ...data,
+      sequence_no: nextSequenceNo,
+      created_at: currentDate,
+      updated_at: currentDate,
+    },
+  })
+  return { message: "Successfully added", data: newCategory }
+}
+
+export const update = async (id: number, data: Prisma.skill_categoriesUpdateInput) => {
+  const updatedSkillCategory = await prisma.skill_categories.update({
+    where: {
+      id,
+    },
+    data: {
+      ...data,
+      updated_at: new Date(),
+    },
+  })
+  return { message: "Successfully updated", data: updatedSkillCategory }
+}
+
+export const destroy = async (id: number) => {
+  const skillCategory = await prisma.skill_categories.findFirst({
+    where: {
+      id,
+    },
+  })
+  await prisma.skill_categories.delete({
+    where: {
+      id: skillCategory?.id,
+    },
+  })
+  const remainingSkillCategories = await prisma.skill_categories.findMany()
+
+  for (const category of remainingSkillCategories) {
+    if (category.sequence_no !== null && category.sequence_no !== undefined) {
+      await prisma.skill_categories.update({
+        where: { id: category.id },
+        data: { sequence_no: remainingSkillCategories.indexOf(category) + 1 },
+      })
+    }
+  }
+}
+export const isUsedInSkillsTable = async (id: number) => {
```

Comment:
Move this as a service inside `skill-category-service`

Reply:
isn't the query is inside the repository?

Reply:
Up, kindly remove this in this repository